src/core/manager.c

```diff
@@ -1591,6 +1591,32 @@ static unsigned manager_dispatch_stop_when_bound_queue(Manager *m) {
         return n;
 }
 
+static unsigned manager_dispatch_stop_notify_queue(Manager *m) {
+        unsigned n = 0;
+
+        assert(m);
+
+        if (m->may_dispatch_stop_notify_queue < 0)
+                m->may_dispatch_stop_notify_queue = hashmap_isempty(m->jobs);
+
+        if (!m->may_dispatch_stop_notify_queue)
+                return 0;
+
+        m->may_dispatch_stop_notify_queue = false;
+
+        LIST_FOREACH(stop_notify_queue, u, m->stop_notify_queue) {
+                assert(u->in_stop_notify_queue);
+
+                assert(UNIT_VTABLE(u)->stop_notify);
+                if (UNIT_VTABLE(u)->stop_notify(u)) {
+                        assert(!u->in_stop_notify_queue);
+                        n++;
+                }
+        }
```

Comment:
i am kinda afraid of corruption  here if the callback adds/removes other entries somehow

Reply:
Yeah, but I don't see any immediately better option (without introducing a generation for this, that is, but seems like an overkill with this sole user of it)